src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -181,34 +184,44 @@ class JesBackend extends Backend with LazyLogging {
     val redirectedCommand = s"$instantiatedCommandLine > $LocalStdoutValue 2> $LocalStderrValue"
 
-    val status = Pipeline(redirectedCommand, workflowDescriptor, call, jesParameters, GoogleProject, JesConnection).run.waitUntilComplete()
+    val run = Pipeline(redirectedCommand, workflowDescriptor, call, jesParameters, GoogleProject, JesConnection).run
+    // Wait until the job starts (or completes/fails) before registering the abort to avoid awkward cancel-during-initialization behavior.
+    run.waitUntilRunningOrComplete()
+    abortFunctionRegistration.registrationFunction apply AbortFunction({() => run.abort()})
+    try {
+      val status = run.waitUntilComplete()
 
-    def taskOutputToRawValue(taskOutput: TaskOutput): Try[WdlValue] = {
-      val jesOutputFile = unsafeJesOutputs find {_.name == taskOutput.name} map {j => WdlFile(j.gcs)}
-      jesOutputFile.map(Success(_)).getOrElse(taskOutput.expression.evaluate(scopedLookupFunction, engineFunctions, interpolateStrings = true))
-    }
-
-    val outputMappings = call.task.outputs.map { taskOutput =>
-      val rawValue = taskOutputToRawValue(taskOutput)
-      logger.debug(s"JesBackend setting ${taskOutput.name} to $rawValue")
-      taskOutput.name -> rawValue
-    }.toMap
+      def taskOutputToRawValue(taskOutput: TaskOutput): Try[WdlValue] = {
+        val jesOutputFile = unsafeJesOutputs find { _.name == taskOutput.name } map { j => WdlFile(j.gcs) }
+        jesOutputFile.map(Success(_)).getOrElse(taskOutput.expression.evaluate(scopedLookupFunction, engineFunctions, interpolateStrings = true))
+      }
 
-    val warnAboutStderrLength: BigInteger =
-      if (!call.failOnStderr) BigInteger.valueOf(0)
-      else JesConnection.storage.objectSize(GoogleCloudStoragePath(stderrJesOutput(callGcsPath).gcs))
+      def processOutput(taskOutput: TaskOutput, processingFunction: TaskOutput => Try[WdlValue]) = {
+        val rawValue = processingFunction(taskOutput)
+        logger.debug(s"JesBackend setting ${taskOutput.name} to $rawValue")
+        taskOutput.name -> rawValue
+      }
 
-    if (warnAboutStderrLength.intValue > 0) {
-      Failure(new Throwable(s"Workflow ${workflowDescriptor.id}: stderr has length $warnAboutStderrLength for command: $instantiatedCommandLine"))
-    } else {
-      status match {
-        //
-        case Run.Success(created, started, finished) =>
-          unwrapOutputValues(outputMappings, workflowDescriptor)
-        case Run.Failed(created, started, finished, errorCode, errorMessage) =>
-          Failure(new Throwable(s"Workflow ${workflowDescriptor.id}: errorCode $errorCode for command: $instantiatedCommandLine. Message: $errorMessage"))
+      val outputMappings = call.task.outputs.map {taskOutput => processOutput(taskOutput, taskOutputToRawValue)}.toMap
+
+      lazy val warnAboutStderrLength: BigInteger = JesConnection.storage.objectSize(GoogleCloudStoragePath(stderrJesOutput(callGcsPath).gcs))
```

Comment:
`stderrLength` would be a more accurate name for the variable, and I'm not sure the `BigInteger` type ascription is needed